domain/src/main/java/org/oppia/android/domain/topic/TopicListController.kt

```diff
@@ -161,101 +190,291 @@ class TopicListController @Inject constructor(
+    playState: ChapterPlayState
+  ): List<ChapterProgress> {
+    return storyProgress.chapterProgressMap.values
+      .filter { chapterProgress -> chapterProgress.chapterPlayState == playState }
+      .sortedByDescending { chapterProgress -> chapterProgress.lastPlayedTimestamp }
+  }
+
+  private fun createOngoingStoryListBasedOnRecentlyPlayed(
+    storyId: String,
+    story: StorySummary,
+    recentlyPlayerChapterProgress: ChapterProgress,
+    completedChapterProgressList: List<ChapterProgress>,
+    topic: Topic,
+    isTopicConsideredCompleted: Boolean
+  ): Pair<PromotedStory?, Long> {
+    var numberOfDaysPassed = 0L
+    val recentlyPlayerChapterSummary: ChapterSummary? =
+      story.chapterList.find { chapterSummary ->
+        recentlyPlayerChapterProgress.explorationId == chapterSummary.explorationId
+      }
+    if (recentlyPlayerChapterSummary != null) {
+      numberOfDaysPassed = recentlyPlayerChapterProgress.getNumberOfDaysPassed()
+      return Pair(
+        addPromotedStoryInRecommendedStoryList(
+          storyId,
+          topic,
+          completedChapterProgressList.size,
+          story.chapterCount,
+          recentlyPlayerChapterSummary.name,
+          recentlyPlayerChapterSummary.explorationId,
+          isTopicConsideredCompleted
+        ),
+        numberOfDaysPassed
+      )
+    }
+    return Pair(null, numberOfDaysPassed)
+  }
+
+  private fun createOngoingStoryListBasedOnMostRecentlyCompleted(
+    storyId: String,
+    story: StorySummary,
+    mostRecentCompletedChapterProgress: ChapterProgress,
+    completedChapterProgressList: List<ChapterProgress>,
+    topic: Topic,
+    isTopicConsideredCompleted: Boolean
+  ): Pair<PromotedStory?, Long> {
+    var numberOfDaysPassed = 0L
+    val lastChapterSummary: ChapterSummary? =
+      story.chapterList.find { chapterSummary ->
+        mostRecentCompletedChapterProgress.explorationId == chapterSummary.explorationId
+      }
+    val nextChapterIndex = story.chapterList.indexOf(lastChapterSummary) + 1
+    if (story.chapterList.size > nextChapterIndex) {
+      val nextChapterSummary: ChapterSummary? = story.chapterList[nextChapterIndex]
+      if (nextChapterSummary != null) {
+        numberOfDaysPassed = mostRecentCompletedChapterProgress.getNumberOfDaysPassed()
+        return Pair(
+          addPromotedStoryInRecommendedStoryList(
+            storyId,
+            topic,
+            completedChapterProgressList.size,
+            story.chapterCount,
+            nextChapterSummary.name,
+            nextChapterSummary.explorationId,
+            isTopicConsideredCompleted
+          ),
+          numberOfDaysPassed
+        )
+      }
     }
-    return ongoingStoryListBuilder.build()
+    return Pair(null, numberOfDaysPassed)
   }
 
-  private fun createRecommendedStoryList(): List<PromotedStory> {
-    val recommendedStories = ArrayList<PromotedStory>()
+  private fun addPromotedStoryInRecommendedStoryList(
```

Comment:
Can we remove this now? It just seems to be calling ``createPromotedStory``--we should inline it. The name is also misleading--it's not adding anything anywhere.

Reply:
Removed this function.